src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -186,10 +179,25 @@ public void updateFilteredBandList(Predicate<Band> predicate) {
         filteredBands.setPredicate(predicate);
     }
 
+    /**
+     * Returns an unmodifiable view of the list of {@code Band} backed by the internal list of
+     * {@code versionedAddressBook}. Returns an unmodifiable view of the list of {@code Musician}
+     * backed by the internal list of {@code versionedAddressBook}
+     * If there is no band with the corresponding name, an error message will be displayed
+     * And the panel will show initial state of listing all.
+     */
     @Override
-    public void updateFilteredBandList(int bandIndex) {
-        Predicate<Band> predicate = new BandIsSamePredicate(filteredBands.get(bandIndex));
-        filteredBands.setPredicate(predicate);
+    public void updateFilteredBandMusicianList(Predicate<Band> bandPredicate) {
+        filteredBands.setPredicate(bandPredicate);
+
+        if (filteredBands.size() == 0) {
+            updateFilteredMusicianList(PREDICATE_SHOW_ALL_MUSICIANS);
+            updateFilteredBandList(PREDICATE_SHOW_ALL_BANDS);
+            return;
+        }
+
+        Predicate<Musician> musicianPredicate = new MusicianInBandPredicate(filteredBands.get(0));
+        filteredMusicians.setPredicate(musicianPredicate);
```

Comment:
I can see that the filtered musician list and band list are combined here to form a coherent logic since we only list the musicians that are in a specific band. This decision LGTM!